### PR TITLE
Fixes contention bug and adds test for the bug.

### PR DESCRIPTION
**Pull request by ghost**
## Overview
Quantifier [discovered a bug](https://discordapp.com/channels/391435113415573504/420661762828926978/675797443660480543) in TreeHashMap.  I replicated the bug and found it was due to a get-then-update rather than an atomic update.  This fixes that bug.

### CODE REVIEW

File: casper/src/main/resources/Registry.rho [hunk 1]
```diff
@@ -188,28 +152,64 @@ in {
...
-                  // Node missing
-                  // Create node, set node to 0
-                  MakeNode!(0, (map, nybList.slice(0, n + 1))) |
-                  new ack in {
-                    // Atomically update current node to val | (1 << nybList.nth(n))
-                    nodeSetBit!((map, nybList.slice(0, n)), nybList.nth(n), *ack) |
-                    for (_ <- ack) {
+            match (map, nybList.slice(0, n)) {
+              node => {
+                for (@val <<- @[node, *storeToken]) {
```
Comment (Isaac-DeFrain) [inline casper/src/main/resources/Registry.rho:157]: Nice. Now the lock is only acquired if there's a value present at the node. Cool.

**Review by Isaac-DeFrain — APPROVED** [review]
LGTM

**Review by tgrospic — APPROVED** [review]
Excellent! With this fix PoSTest in @Isaac-DeFrain PR #2875 now passes.

**Pull request merged**